hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -471,7 +470,8 @@ public void remove() {
   private static FSDataInputStream getFSDataInputStream(FileSystem fs,
                                                         HoodieLogFile logFile,
                                                         int bufferSize) throws IOException {
-    FSDataInputStream fsDataInputStream = fs.open(logFile.getPath(), bufferSize);
+    String escapePathName = PartitionPathEncodeUtils.unescapePathName(logFile.getPath().toString());
+    FSDataInputStream fsDataInputStream = fs.open(new Path(escapePathName), bufferSize);
```

Comment:
this has performance impact; the unescape is looking at every char, also calling `logFile.getPath()` construct a Path object and then L474 constructs another Path, which is costly. There should be a deeper fix to this: we shouldn't have to unescape every `logFile.getPath()` to read.